src/opnsense/mvc/app/controllers/OPNsense/Firewall/Api/AliasUtilController.php

```diff
@@ -132,13 +165,16 @@ public function flushAction($alias)
     /**
      * delete item from alias table
      * @param string $alias name
+     * @param $address
      * @return array status
+     * @throws \Exception
      */
-    public function deleteAction($alias)
+    public function deleteAction($alias, $address)
```

Comment:
Just to be sure, did you test this with different types of items (ip[4|6] address, network)? 
This could cause issues with escaping, which is the reason why the address was initially send with the post itself (+ easy to extent the call to support multiple addresses at ones) 

We probably better keep this call like it was, people might already be using this endpoint in other scripts/applications too.

Reply:
I ran into the escaping issue, so I encoded the address with `urlencode`. I have tested it with regular IPv4 addresses and ranges like `192.168.1/24`. I haven't tested it with IPv6, though, but since `urlencode` handles `:`, it should not cause any surprises.

I would have preferred not touching this, but the `del` action on Bootgrid sends this value as a GET parameter, with an empty POST body. I have tried working around it, but it seems that I would have to bypass the `del` action entirely. I actually tried doing so, but I ran into a race issue with that.

The ValueA framework has a nice way of adding custom commands, but that does not appear to be present here. If it were to be backported, it would provide an easy way out.

What would be the best course of action, you think?

Reply:
I see, hooking custom functions should be rather simple, if I'm not mistaken this  does something similar:

https://github.com/opnsense/core/blob/e0692a85b32239b5d6edbab742b7ed073034d7b6/src/opnsense/mvc/app/views/OPNsense/IDS/index.volt#L247-L249

Better to revert to the original call then.

Reply:
It doesn't look like that function does what I need - it only adds a custom button (which I had no trouble with), but it still uses the default edit action (it has a `command-edit` class, which means that Bootgrid automatically adds the default edit action to it). My problem was that I need to define a custom action, because the default delete action does not work with this call.

Reply:
I found a solution.